jujuclient/mem.go

```diff
@@ -351,13 +361,17 @@ func (c *MemStore) UpdateAccount(controllerName string, details AccountDetails)
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	oldDetails, ok := c.Accounts[controllerName]
+	if ok && c.ImmutableAccount {
```

Comment:
If we created the MemStore using `NewEmbeddedMemStore`, the consumer should never try to call `UpdateAccount()`. So should we reject and return an error instead?

Reply:
The caller doesn't know what type of store it is. Normally, it will be a file store and we do want the update. But the caller has no way of knowing how the store has been set up. So the place that does know can signal that updates should be ignored.

Reply:
ic, fair enough